src/app/dspace/services/bitstream-uploader.service.ts

```diff
@@ -0,0 +1,101 @@
+
+import { DSpaceService } from '../services/dspace.service';
+import { NotificationService } from '../../utilities/notification/notification.service';
+import { Item } from "../models/item.model";
+
+import { TranslateService } from "ng2-translate/ng2-translate";
+import { FileUploader } from 'ng2-file-upload';
+
+/**
+ * A basic bitstream uploader service which extends the default
+ * FileUploader provided by https://github.com/valor-software/ng2-file-upload/
+ * <p>
+ * This service overrides a few key methods to handle uploading of DSpace
+ * bitstreams. See below for more info
+ */
+export class BitstreamUploader extends FileUploader {
+
+    /**
+     * Reference to DSpace Item this bitstream will be uploaded to
+     * This will be set *after* the Item is created, which is why it need to be public
+     */
+    public item: Item;
+
+    /**
+     * Reference to DSpace REST API authentication token.
+     * Required to be set for upload to be authenticated.
+     */
+    public authToken: string;
+
+    /**
+     * Override default constructor to also include key DSpace services
+     * @param translate
+     *      TranslateService
+     * @param dspaceService
+     *      DSpaceService is a singleton service to interact with the dspace service.
+     * @param notificationService
+     *      NotificationService is a singleton service to notify user of alerts.
+     * @param options
+     *      FileUploader options to pass to default FileUploader
+     */
+    constructor(private translate: TranslateService,
```

Comment:
Just fixed this in a new commit. TranslateService is now being used to provide i18n to error notifications